js/filters.js

```diff
@@ -115,9 +145,27 @@ function TypeaheadFilter(elm) {
   var key = this.$body.data('dataset');
   var dataset = typeahead.datasets[key];
   this.typeaheadFilter = new typeaheadFilter.TypeaheadFilter(this.$body, dataset);
```

Comment:
Just to be clear on this line and the intent here (because the `new` keyword in JS always jumps out at me), this is deliberately setting `this.typeaheadFilter` to be the instance of the object created and returned by `typeaheadFilter.TypeaheadFilter` itself, right?

This makes sense in the context of the next line, but I just wanted to be sure that was the intent here.